server/topic.go

```diff
@@ -241,23 +241,33 @@ func (t *Topic) userIsReader(uid types.Uid) bool {
 	return (modeGiven & modeWant).IsReader()
 }
 
-// maybeFixTopicName sets the topic field in `msg` depending on the uid.
-func (t *Topic) maybeFixTopicName(msg *ServerComMessage, uid types.Uid) {
+// maybeFixTopicName sets the topic field in `msg` depending on the uid and subscription type.
+func (t *Topic) maybeFixTopicName(msg *ServerComMessage, uid types.Uid, isChanSub bool) {
 	// For zero uids we don't know the proper topic name.
 	if uid.IsZero() {
```

Comment:
Not sure is `uid` could be zero for channels, but this check is only valid for p2p topics. So, maybe make the first check
`if (t.cat == types.TopicCatP2P && !uid.IsZero()) || (t.cat == types.TopicCatGrp && t.isChan) {`

Reply:
Done.